chainstate/test-suite/src/tests/fungible_tokens_v1.rs

```diff
@@ -0,0 +1,2141 @@
+    utils::concurrency::model(move || {
+        let mut rng = make_seedable_rng(seed);
+        let mut tf = TestFramework::builder(&mut rng).build();
+
+        let token_min_supply_change_fee =
+            tf.chainstate.get_chain_config().token_min_supply_change_fee();
+
+        let amount_to_mint =
+            Amount::from_atoms(rng.gen_range(2..SignedAmount::MAX.into_atoms() as u128));
+        let amount_to_burn = Amount::from_atoms(rng.gen_range(1..amount_to_mint.into_atoms()));
+
+        let (token_id, _, utxo_with_change) =
+            issue_token_from_genesis(&mut rng, &mut tf, TokenTotalSupply::Unlimited);
+
+        let best_block_id = tf.best_block_id();
+        let (_, mint_tx_id) = mint_tokens_in_block(
+            &mut tf,
+            best_block_id,
+            utxo_with_change,
+            token_id,
+            amount_to_mint,
+            true,
+        );
+
+        let nonce = BlockchainStorageRead::get_account_nonce_count(
+            &tf.storage,
+            AccountType::TokenSupply(token_id),
+        )
+        .unwrap()
+        .map_or(AccountNonce::new(0), |n| n.increment().unwrap());
+
+        // it's ok to burn less tokens than the input has. In this case only the burned amount will be unminted
+        tf.make_block_builder()
+            .add_transaction(
+                TransactionBuilder::new()
+                    .add_input(
+                        TxInput::from_account(
+                            nonce,
+                            AccountSpending::TokenCirculatingSupply(token_id),
+                        ),
+                        InputWitness::NoSignature(None),
+                    )
+                    .add_input(
+                        TxInput::from_utxo(mint_tx_id.into(), 1),
+                        InputWitness::NoSignature(None),
+                    )
+                    .add_input(
+                        TxInput::from_utxo(mint_tx_id.into(), 2),
+                        InputWitness::NoSignature(None),
+                    )
+                    .add_output(TxOutput::Burn(OutputValue::Coin(
+                        token_min_supply_change_fee,
+                    )))
+                    .add_output(TxOutput::Burn(OutputValue::TokenV1(
+                        token_id,
+                        amount_to_burn,
+                    )))
+                    .build(),
+            )
+            .build_and_process()
+            .unwrap();
+
+        let actual_supply =
+            TokensAccountingStorageRead::get_circulating_supply(&tf.storage, &token_id).unwrap();
+        assert_eq!(
+            actual_supply,
+            Some((amount_to_mint - amount_to_burn).unwrap())
+        );
+    });
+}
+
+// Issue and mint some tokens.
+// On unmint provide a utxo that has less tokens that burned output.
+// Check that it's an error.
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn try_to_burn_less_by_providing_smaller_input_utxo(#[case] seed: Seed) {
```

Comment:
Isn't that "burn more" than allowed?

Reply:
"burning more" than provided by input is invalid because it's money printing error

Reply:
OK. Point taken.

Please remove the try_to prefix.